Mage.Sets/src/mage/cards/t/ThePrideOfHullClade.java

```diff
@@ -0,0 +1,151 @@
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.constants.CostModificationType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.abilities.keyword.DefenderAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class ThePrideOfHullClade extends CardImpl {
+
+    public ThePrideOfHullClade(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{10}{G}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.CROCODILE);
+        this.subtype.add(SubType.ELK);
+        this.subtype.add(SubType.TURTLE);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(15);
+
+        // This spell costs {X} less to cast, where X is the total toughness of creatures you control.
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new ThePrideOfHullCladeCostReductionEffect()));
+
+        // Defender
+        this.addAbility(DefenderAbility.getInstance());
+
+        // {2}{U}{U}: Until end of turn, target creature you control gets +1/+0, gains "Whenever this creature deals combat damage to a player, draw cards equal to its toughness," and can attack as though it didn't have defender.
+        Ability ability = new SimpleActivatedAbility(
+                new BoostTargetEffect(1, 0)
+                        .setText("Until end of turn, target creature you control gets +1/0"),
+                new ManaCostsImpl<>("{2}{U}{U}"))
+                ;
+        ability.addEffect(
+                new GainAbilityTargetEffect(
+                        new DealsCombatDamageToAPlayerTriggeredAbility(
+                                new ThePrideOfHullCladeEffect(), false))
+                        .setText(", gains \"Whenever this creature deals combat damage to a player, draw cards equal to its toughness,\""));
+        ability.addEffect(
+                new CanAttackAsThoughItDidntHaveDefenderTargetEffect(Duration.EndOfTurn)
+                        .setText("and can attack as though it didn't have defender."));
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private ThePrideOfHullClade(final ThePrideOfHullClade card) {
+        super(card);
+    }
+
+    @Override
+    public ThePrideOfHullClade copy() {
+        return new ThePrideOfHullClade(this);
+    }
+}
+
+class ThePrideOfHullCladeCostReductionEffect extends CostModificationEffectImpl {
```

Comment:
is there a reason to use a custom class rather than `SpellCostReductionSourceEffect` with a DynamicValue?